src/app/task/task.component.ts

```diff
@@ -20,16 +20,20 @@ export class TaskComponent implements OnInit {
 
   ngOnInit() { }
 
+  getId(min, max) {
```

Comment:
Id should be generated in Task class constructor when new object is created.

Reply:
Do you mean I have to call this function first time in component? or definition of it has to be there?

Reply:
It is in the component right now. It's just that front end shouldn't be responsible for generating id. I consider component file in as frontend in our case. Usually when you send post request and object has id it usually means that something is wrong. So, let's move in different place. Let's move genId method to the class where our list of tasks lives and invoke it there to set Task id before adding task to the list.